Python/ceval.c

```diff
@@ -1064,9 +1064,11 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
            Py_MakePendingCalls() above. */
 
         if (_Py_atomic_load_relaxed(&eval_breaker)) {
-            if (_Py_OPCODE(*next_instr) == SETUP_FINALLY) {
+            if (_Py_OPCODE(*next_instr) == SETUP_FINALLY
+                || _Py_OPCODE(*next_instr) == YIELD_FROM) {
```

Comment:
@AraHaan: I don't think I understand... Can you elaborate on what you're worried about? The bug we're trying to fix here is that *while* the `yield from` is running, `KeyboardInterrupt` should only be raised inside `generator2`, not `generator1`.

I guess there is a small annoyance that if a `KeyboardInterrupt` arrives at just the wrong moment we could get a spurious "coroutine was not awaited" warning, but even I can't get *too* worked up about that...